src/main/java/commands/workout/workoutsession/WorkoutSessionSearch.java

```diff
@@ -1,37 +1,53 @@
 package commands.workout.workoutsession;
 
 import commands.Command;
+import models.Exercise;
 import storage.workout.Storage;
 import ui.workout.workoutsession.WorkoutSessionUi;
 import workout.workoutsession.WorkoutSessionParser;
-import workout.workoutsession.exercise.ExerciseList;
+import models.ExerciseList;
+
+import java.util.ArrayList;
+
+import static ui.CommonUi.LS;
 
 
 public class WorkoutSessionSearch extends Command {
+    private boolean isEmptySearchResult = true;
 
     @Override
     public void execute(String[] inputs, ExerciseList exerciseList,
                         String filePath, Storage storage, boolean[] hasEndedWorkoutSessions) {
         String searchTerm = WorkoutSessionParser.searchParser(inputs).toLowerCase();
-        StringBuilder searchResult = new StringBuilder();
 
         if (searchTerm.length() > 0) {
 
-            for (int i = 0; i < exerciseList.exerciseList.size(); i++) {
+            String searchResult = formatList(exerciseList.exerciseList, searchTerm);
 
-                if (exerciseList.exerciseList.get(i).getDescription().toLowerCase().contains(searchTerm)) {
-                    searchResult.append(exerciseList.exerciseList.get(i).toString() + "\n\t ");
-                }
-            }
-
-            if (searchResult.length() != 0) {
-                ui.showToUser(searchResult.toString().trim());
+            if (!isEmptySearchResult) {
+                ui.showToUser(searchResult);
             } else {
                 WorkoutSessionUi.searchResultsEmpty();
             }
-
         } else {
             WorkoutSessionUi.searchInputError();
         }
     }
+
+    private String formatList(ArrayList<Exercise> exercise, String searchTerm) {
+        String returnString = String.format("%-8s", "Index") + String.format("%-20s", "Exercise")
```

Comment:
This format is also repeated in WorkoutSessionList, so you can consider refactoring it to a constant.